src/integrators/controllers.jl

```diff
@@ -136,15 +136,139 @@ end
 
 function stepsize_controller!(integrator, alg::QNDF)
   cnt = integrator.iter
+  EEst1 = integrator.cache.EEst1
+  EEst2 = integrator.cache.EEst2
   if cnt <= 3
-    q = standard_stepsize_controller!(integrator, alg)
+    # q = standard_stepsize_controller!(integrator, alg)
+    q = 1 #quasi-contsant steps    
   else
-    q = integrator.dt/integrator.cache.h
-    integrator.qold = integrator.dt/q
+    dt_optim_success, dt_optim_failed = QNDF_stepsize_and_order!(integrator.cache, integrator.EEst, EEst1, EEst2, integrator.dt, integrator.cache.order)
+    q = integrator.dt/dt_optim_success
+    integrator.qold = integrator.dt/dt_optim_failed
   end
   q
 end
 
+function step_accept_controller!(integrator,alg::QNDF,q)
+  #step is accepted, reset count of consecutive failed steps
+  integrator.cache.c = 0
```

Comment:
we should have a better variable name for number of failed steps. It's fine if the paper uses `c` and we can make this change later

Reply:
Made the requested changes.